File: app/src/main/java/net/ginteam/carmen/network/api/service/CityService.java
package net.ginteam.carmen.network.api.service;

import net.ginteam.carmen.model.ResponseModel;
import net.ginteam.carmen.model.city.CityModel;
import net.ginteam.carmen.network.api.ApiLinks;

import java.util.List;

import retrofit2.http.GET;
import retrofit2.http.Query;
import rx.Observable;

/**
 * Created by Eugene on 12/27/16.
 */

public interface CityService {

    @GET(ApiLinks.CATALOG.CITIES)
    Observable <ResponseModel <List<CityModel>>> fetchCities();

    @GET(ApiLinks.CATALOG.CITIES_BY_POINT)
    Observable <ResponseModel <CityModel>> fetchCityByPoint(
            @Query(ApiLinks.CATALOG.LAT) Double lat,
            @Query(ApiLinks.CATALOG.LNG) Double lon
    );


}

File: app/src/main/kotlin/net/ginteam/carmen/kotlin/api/service/CityService.kt
package net.ginteam.carmen.kotlin.api.service

import net.ginteam.carmen.kotlin.api.ApiSettings
import net.ginteam.carmen.kotlin.model.CityModel
import net.ginteam.carmen.kotlin.model.ResponseModel
import retrofit2.http.GET
import rx.Observable

/**
 * Created by eugene_shcherbinock on 2/9/17.
 */
interface CityService {

    @GET(ApiSettings.Catalog.GET_CITIES)
    fun fetchCities(): Observable <ResponseModel <List <CityModel>>>

    @GET(ApiSettings.Catalog.GET_CITY_BY_POINT)
    fun fetchCity(): Observable <ResponseModel <CityModel>>

}